eval/unwrap.go

```diff
@@ -102,7 +108,20 @@ func (u ValueUnwrapper) FdOrClose() (int, error) {
 	if err == nil && s == "-" {
 		return -1, nil
 	}
-	return u.NonNegativeInt()
+	switch s {
```

Comment:
You can delegate all this work to the `Fd` method.

Also put the `Fd` method just above or under this method as they are closely related.